MERP_Character_Sheet_BE/Repository/GameClassRepository.cs

```diff
@@ -51,23 +52,42 @@ public async Task<bool> Update(GameClass _class)
             return success;
         }
 
-        public GameClass Get(long classId)
+        public GameClassDTO Get(long classId)
+        {
+            var result = _dbContext.Classes
+                                .Where(x => x.Id == classId)
+                                .FirstOrDefault();
+            if(result == null)
+            {
+                return null;
+            }
+            return (GameClassDTO)result;
+        }
+
+        public GameClass GetWithIDs(long classId)
         {
             return _dbContext.Classes
                                 .Where(x => x.Id == classId)
                                 .FirstOrDefault();
         }
 
-        public IOrderedQueryable<GameClass> GetAll()
+        public List<GameClassDTO> GetAll()
         {
-            return _dbContext.Classes.OrderByDescending(x => x.Id);
+            var parsedList = new List<GameClassDTO>();
+            var unparsed = _dbContext.Classes.OrderByDescending(x => x.Id).ToList();
+            
+            unparsed.ForEach(delegate(GameClass raw){
```

Comment:
this parsing using delegates and explicit casts seems a big dark magic lol, I'm just wondering if it's worth the effort since for a regular parser you'll need to add the fields manually in a static parser class and in here probably would be the same in the explicit operator, but seems different thumbs up

Reply:
As far as what I understood, the delegate reference is the equivalent to have an anonymous fn in JS :)